model_analyzer/record/metrics_manager.py

```diff
@@ -279,13 +280,14 @@ def _get_measurement_if_config_duplicate(self, run_config):
         results = self._state_manager.get_state_variable(
             'ResultManager.results')
 
-        # check whether perf config string is a key in result dict
-        if model_name not in results:
+        if not results.contains_model(model_name):
             return False
-        if not self._is_config_in_results(
-                run_config.model_config()._model_config, results[model_name]):
+
+        if not results.contains_model_config(model_name, model_config_name):
             return False
-        measurements = results[model_name][model_config_name][1]
+
+        measurements = results.get_model_config_measurements_dict(
```

Comment:
What is `measurements` here? A list of measurements? A dict?

Reply:
It's a dict (see the results function). This is all going to change when the new measurement class gets implemented, so I didn't feel like putting effort into rewriting this as part of this story.

Reply:
nevermind I thought we were changing the return type of the function